dashboard/app/models/script.rb

```diff
@@ -1437,10 +1439,10 @@ def finish_url
   # A script that the general public can assign. Has been soft or
   # hard launched.
   def launched?
-    [SharedConstants::PUBLISHED_STATE.preview, SharedConstants::PUBLISHED_STATE.stable].include?(published_state)
+    [SharedConstants::PUBLISHED_STATE.preview, SharedConstants::PUBLISHED_STATE.stable].include?(get_published_state)
   end
 
-  def published_state
+  def get_published_state
```

Comment:
What is the idea behind this method -- is it just to get the initial published state as part of the move to published_state?

Reply:
This is method is just a temporary method to transition. It is removed in the [next PR](https://github.com/code-dot-org/code-dot-org/pull/41027). I renamed it in this PR to make it easier to tell the difference between it and in the new field.

Reply:
> I know you expressed in the doc that you were interested in pilot just being a kind of in-development state. I'm open it discussing and exploring this. To me, they are different states and I feel like they could grow independent of each other. In development will be something only levelbuilders can see. It guarantees me that the thing I am working on is not available to anyone else. The pilot state is what I move something to once I want to allow a limited set of teachers to use it. When I pick the pilot state I am able to then set the name of the pilot where I will add teachers. I could see us wanting to allow some feature while piloting but not while something is just in development which is why I think it's safer for them to be separate.

OK - separate states is fine with me.

Reply:
OK, so I am seeing the following in this PR:
* script.get_published_state returns published state based on existing booleans
* migration to set the new script.published_state column using the script.get_published_state method
* no production code actually using script.published_state yet. instead, it uses script.get_published_state

Does that sound accurate? 

I don't see a specific problem with this, but it is a lot to be in one PR, especially with too many commits in the history to be readable individually. For next time I would recommend something like:
1. rename method to get_published_state
2. migration to populate published_state field
3. everything else

No need to go back and change it now -- but given the nature of the change, this would help not just with reviewability but also with unwinding things if there is a problem with any of the steps.

Reply:
I totally agree that this was not the ideal way to have this PR set up. I was trying to go fast to get as much done before leaving and ended up with a messy chain of PRs that only got worse as things changed in the earlier ones. Thank you for reviewing it with this added complication.